elfi/examples/gauss.py

```diff
@@ -1,29 +1,90 @@
-"""An example implementation of a Gaussian noise model."""
+"""The general, n-dimensional Gaussian noise model."""
 
 from functools import partial
 
 import numpy as np
 import scipy.stats as ss
 
 import elfi
+from elfi.examples.gnk import euclidean_multidim
 
 
-def Gauss(mu, sigma, n_obs=50, batch_size=1, random_state=None):
+def Gauss(*params, cov_ii=None, cov_ij=None, n_obs=15, batch_size=1, n_dim, random_state=None):
     """Sample the Gaussian distribution.
 
+    Reference
+    ---------
+    The default settings replicate the experiment settings used by:
+        [1] arXiv:1704.00520 (Järvenpää et al., 2017).
+
     Parameters
     ----------
-    mu : float, array_like
-    sigma : float, array_like
+    *params : array_like
+        The first (n - 2) elements corresponds to the mean parameters,
+        and the last two parameters correspond to the variance parameters.
```

Comment:
`cov_ii` and `cov_ij` are used if one wants to avoid inferring the variance.

I could improve the implementation by dedicating a separate parameter for the diagonal covariance in each dimension. However, in this case, do we leave the non-diagonal covariance terms same? In the alternative case, say upon the inference of the n-dimensional Gaussian noise model, we would have n^2 - n parameters purely for the non-diagonal covariance .